src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -44,43 +41,52 @@ internal sealed class MemoryMappedInfo : IDisposable
             /// However, the operating system does not actually close the views which are in use until the view handles
             /// are closed as well, even if the <see cref="MemoryMappedFile"/> is disposed first.</para>
             /// </remarks>
-            private readonly MemoryMappedFile _memoryMappedFile;
+            private readonly ReferenceCountedDisposable<MemoryMappedFile> _memoryMappedFile;
 
             /// <summary>
-            /// actual memory accessor that owns the VM
+            /// A weak reference to the actual memory accessor that owns the VM
             /// </summary>
             /// <remarks>
-            /// <para>It is possible for this accessor to be disposed prior to the streams which use it. However, the
-            /// streams interact directly with the underlying memory buffer, and keep a
-            /// <see cref="CopiedMemoryMappedViewHandle"/> to prevent that buffer from being released while still in
-            /// use. The <see cref="SafeHandle"/> used by this accessor is reference counted, and is not finally
-            /// released until the reference count reaches zero.</para>
+            /// <para>This holds a weak counted reference to current <see cref="MemoryMappedViewAccessor"/>, which
+            /// allows additional accessors for the same address space to be obtained up until the point when no
+            /// external code is using it. When the memory is no longer being used by any
+            /// <see cref="SharedReadableStream"/> objects, the view of the memory mapped file is automatically
+            /// unmapped, making the process address space it previously claimed available for other purposes. If/when
+            /// it is needed again, a new view is created.</para>
             /// </remarks>
-            private MemoryMappedViewAccessor _accessor;
+            private ReferenceCountedDisposable<MemoryMappedViewAccessor>.WeakReference _weakAccessor;
```

Comment:
Sorry, i've forgotten now.  Why do we need this?  Aren't all references to the memory mapped data passed out through streams?  All our clients should dispose those streams.  So why can't we deterministically release the data once all the streams are gone?